src/selectors/profile.js

```diff
@@ -727,3 +727,18 @@ export const getProfileFilterPageData: Selector<ProfileFilterPageData | null> =
   getRelevantInnerWindowIDsForCurrentTab,
   extractProfileFilterPageData
 );
+
+/**
+ * Get the map of Thread ID -> Thread Name for easy access.
+ */
+export const getThreadIdToNameMap: Selector<
+  Map<number, string>
+> = createSelector(getThreads, threads => {
+  const threadIdToNameMap = new Map();
+  for (const thread of threads) {
+    if (thread.tid !== undefined) {
+      threadIdToNameMap.set(thread.tid, thread.name);
+    }
+  }
+  return threadIdToNameMap;
+});
```

Comment:
@julienw hey, this is still a draft PR but I wanted to get some early feedback about the possible performance impact of this selector. So we are using this selector for only FileIO markers but we have to calculate this data all the time when we show a marker tooltip since this selector is being called inside the TooltipMarker component. I think this should be pretty fast to iterate over the threads since it shouldn't be too many (more than a hundred is really hard capture either and it's pretty rare.) It should also not have a big memory impact. What do you think about this? Should we try to delay this selector call more, like just before the need of this data?

Reply:
I think it's fine because:
* as you said, we don't have that many threads
* this is run only once! Because the `threads` value never changes (unless we publish and sanitize, or revert to a previous profile, which is very rare). So it will be cached once and reused then.

So don't worry here :-) 

It's good to worry about selectors performance in general, so thanks for asking!

Reply:
Thanks for the feedback! I've updated the code with tests, and now it's ready for the review :)